File: 백준/Silver/10845. 큐/큐.py
import sys
from collections import deque

input = sys.stdin.readline

N = int(input())
queue = deque()
result = list()

for _ in range(N):
    line = list(input().strip().split())
    if line[0] == "push":
        queue.append(int(line[1]))
    elif line[0] == "pop":
        if queue:
            result.append(queue.popleft())
        else:
            result.append(-1)
    elif line[0] == "size":
        result.append(len(queue))
    elif line[0] == "empty":
        if queue:
            result.append(0)
        else:
            result.append(1)
    elif line[0] == "front":
        if queue:
            result.append(queue[0])
        else:
            result.append(-1)
    elif line[0] == "back":
        if queue:
            result.append(queue[-1])
        else:
            result.append(-1)

sys.stdout.write('\n'.join(map(str, result)) + '\n')

File: 백준/Silver/10845. 큐/큐.java
import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
        int N = Integer.parseInt(br.readLine().trim());

        Deque<Integer> queue = new LinkedList<>();
        StringBuilder sb = new StringBuilder();

        for(int i = 0; i < N; i++){
            String[] line = br.readLine().split(" ");
            if(line[0].equals("push")){
                queue.offerLast(Integer.parseInt(line[1]));
            }else if(line[0].equals("pop")){
                if (queue.isEmpty()) sb.append(-1).append('\n');
                else sb.append(queue.pollFirst()).append('\n');
            }else if(line[0].equals("size")){
                sb.append(queue.size()).append('\n');
            }else if(line[0].equals("empty")){
                if (queue.isEmpty()) sb.append(1).append('\n');
                else sb.append(0).append('\n');
            }else if(line[0].equals("front")){
                if (queue.isEmpty()) sb.append(-1).append('\n');
                else sb.append(queue.peekFirst()).append('\n');
            }else if(line[0].equals("back")){
                if (queue.isEmpty()) sb.append(-1).append('\n');
                else sb.append(queue.peekLast()).append('\n');
            }
        }

        bw.write(sb.toString());
        bw.flush();
        bw.close();
        br.close();
    }
}